File: rxpanda/src/main/java/com/pandaq/rxpanda/requests/okhttp/GetRequest.java
package com.pandaq.rxpanda.requests.okhttp;

import com.pandaq.rxpanda.RxPanda;
import com.pandaq.rxpanda.observer.ApiObserver;
import com.pandaq.rxpanda.requests.okhttp.base.HttpRequest;
import io.reactivex.Observable;

import java.lang.reflect.Type;

/**
 * Created by huxinyu on 2019/1/11.
 * Email : panda.h@foxmail.com
 * <p>
 * Description : get Request
 */
public class GetRequest extends HttpRequest<GetRequest> {

    public GetRequest(String url) {
        super(url);
    }

    @Override
    protected <T> Observable<T> execute(Type type) {
        return mApi.get(url, localParams)
                .doOnSubscribe(disposable -> {
                    if (tag != null) {
                        RxPanda.manager().addTag(tag, disposable);
                    }
                })
                .compose(httpTransformer(type));
    }

    @SuppressWarnings("unchecked")
    @Override
    protected void execute(ApiObserver callback) {
        if (tag != null) {
            RxPanda.manager().addTag(tag, callback);
        }
        this.execute(getType(callback)).subscribe(callback);
    }
}

File: rxpanda/src/main/java/com/pandaq/rxpanda/requests/okhttp/GetRequest.kt
package com.pandaq.rxpanda.requests.okhttp

import com.pandaq.rxpanda.RxPanda
import com.pandaq.rxpanda.cast
import com.pandaq.rxpanda.requests.okhttp.base.HttpRequest
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import okhttp3.ResponseBody
import java.lang.reflect.Type

/**
 * Created by huxinyu on 2019/1/11.
 * Email : panda.h@foxmail.com
 *
 *
 * Description : get Request
 */
class GetRequest(url: String) : HttpRequest<GetRequest>(url) {

    override suspend fun execute(): ResponseBody? {
        return mApi?.get(url, localParams)
    }

}